src/MinterGateway.sol

```diff
@@ -866,13 +873,22 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
      * @param newTimestamp_ The timestamp of the collateral update.
      */
     function _updateCollateral(address minter_, uint240 amount_, uint40 newTimestamp_) internal {
-        uint40 lastUpdateTimestamp_ = _minterStates[minter_].updateTimestamp;
+        MinterState storage minterState_ = _minterStates[minter_];
 
-        // MinterGateway already has more recent collateral update
+        uint40 lastUpdateTimestamp_ = minterState_.updateTimestamp;
+
+        // Revert if minter's last collateral update is more recent.
         if (newTimestamp_ <= lastUpdateTimestamp_) revert StaleCollateralUpdate(newTimestamp_, lastUpdateTimestamp_);
 
-        _minterStates[minter_].collateral = amount_;
-        _minterStates[minter_].updateTimestamp = newTimestamp_;
+        uint40 latestProposedRetrievalTimestamp_ = minterState_.latestProposedRetrievalTimestamp;
+
+        // Revert if minter's last created retrieval proposal is more recent.
+        if (newTimestamp_ <= latestProposedRetrievalTimestamp_) {
```

Comment:
I checked 2 English dictionaries, stale and obsolete can be considered synonyms. Let's unite line 881 and 886 into 1 check and 1 error.